devcareer/code-jammers-backend
# #175541724 Build CRUD API's for comment

## What does this PR do ##
-  enables user to add a comment
- enables user to delete a comment 
- enables user to edit a comment
- enables user to get a comment

### Description of tasks to be completed? ###
- create add comment controller
- create add comment route
- create get comment controller
- create get comment route
- create edit comment controller
- create edit comment route
- create delete comment controller
- create delete comment route
- write tests for all controllers and route

### How should this be manually tested/checked? ###
- Clone the PR
- Checkout to the branch `ft-user-should-be-able-to-comment-175541724`
- Install the dependencies using `npm install`
- Create a `.env file` and define the following as follows:
   - `DEV_DATABASE_URL = <development database url>`
   - `TEST_DATABASE_URL = <test database url>`
   - `DATABASE_URL = <production database url>`
- run ```npm run migrate``` to create the database tables
- run ```npm run seed``` to seed the database
- Start the server by running ```npm run start-dev```
- Sign in via `http://localhost:3000/api/v1/users/signin` using a `POST` method with
```
     {
          "email": "francis@gmail.com",
          "password": "12345"
     }
```
and copy the token (without the quotes) from the response body.
- You need to sign in with email ```francis@gmail.com``` because that is the user that is the owner of the comments

- paste the token in the authorization header
- make a POST request to the url http://127.0.0.1:3000/api/v1/comment with request body {"comment":"I am a test comment", relatedId:"6003fb36-5112-463e-a1f9-c8944e72412f"}
- check database comment should be added
- make a GET request to url http://127.0.0.1:3000/api/v1/comment/c375c640-81ff-405a-89a8-460ea2f71756
- it should return the comment 
- make a PATCH request to url http://127.0.0.1:3000/api/v1/comment/c375c640-81ff-405a-89a8-460ea2f71756 with request body {"comment":"anything you would like"}
- it should update the comment
- make a delete request to url http://127.0.0.1:3000/api/v1/comment/c375c640-81ff-405a-89a8-460ea2f71756 
- it should delete that comment

### Link to architecture
[architecture](https://docs.google.com/document/d/1FOyxIHMosN-09aP56PWKDxb2Ge3EhNXUs7247dnDbCQ/edit)

### What are the relevant pivotal tracker stories?
[ #175541724](https://www.pivotaltracker.com/story/show/175541724)
